Doc/library/importlib.rst

```diff
@@ -420,30 +420,34 @@ ABC hierarchy::
         reloaded):
 
         - :attr:`__name__`
-            The name of the module.
+            The module's fully-qualified name.
 
         - :attr:`__file__`
-            The path to where the module data is stored (not set for built-in
-            modules).
+            The location of the module's source code.
+            It is ``None`` for namespace packages and not set for built-in
+            modules and frozen modules.
 
         - :attr:`__cached__`
-            The path to where a compiled version of the module is/should be
-            stored (not set when the attribute would be inappropriate).
+            The location of the module's compiled code.
+            It is not set for built-in modules, frozen modules
+            and namespace packages.
 
         - :attr:`__path__`
-            A list of strings specifying the search path within a
-            package. This attribute is not set on modules.
+            The iterable (possibly empty) of the locations of the module's
+            submodules.
+            It is not set for non-package modules.
 
         - :attr:`__package__`
-            The parent package for the module/package. If the module is
-            top-level then it has a value of the empty string. The
-            :func:`importlib.util.module_for_loader` decorator can handle the
-            details for :attr:`__package__`.
+            The module's ``__name__`` attribute for a package, the parent
```

Comment:
I disagree on this, because here is what I observe:

```
$ mkdir -p foo/bar
$ python3 -c "import foo.bar; print(foo.bar.__package__); print(foo.bar.__name__)"
foo.bar
foo.bar
```

And with __init__.py files in each directory it is the same. So for both namespace and regular packages, `package.subpackage.__package__ == package.subpackage.__name__`.

Reply:
You're right.  Packages use their `__name__`.

re-visited in a follow-up review